plugins/resources/compound/libcompound.cpp

```diff
@@ -1817,9 +1834,20 @@ irods::error open_for_prefer_cache_policy(
             irods::log(ret);
             return PASS( ret );
         }
+    }
 
-        if( 0.0 == arch_check_vote ) {
-            *_out_vote = 0.0;
+    // =-=-=-=-=-=-=-
+    // If the archive voted higher it will need to be staged to cache
+    if ( arch_check_vote > cache_check_vote ) {
+
+        // =-=-=-=-=-=-=-
+        // repave the repl requested temporarily
+        irods::file_object_ptr f_ptr = boost::dynamic_pointer_cast< irods::file_object >( _ctx.fco() );
+        int repl_requested = f_ptr->repl_requested();
+        f_ptr->repl_requested( -1 );
```

Comment:
I updated the code to repave repl_requested before calling resolve resource hierarchy on the archive.

I verified that before this change asking for a stale cache replica tried to stage from archive.  Now if you are selecting the cache it will simply return the cache replica even when stale.